custom_components/keymaster/keymaster.yaml

```diff
@@ -207,15 +207,28 @@ binary_sensor:
         {## Get current day of week, date (integer yyyymmdd), and time (integer hhmm) ##}
         {% set current_day = now.strftime('%a')[0:3] | lower %}
         {% set current_date = now.strftime('%Y%m%d') | int %}
+        {% set current_datestamp = as_timestamp(current_date) | int %}
         {% set current_time = now.strftime('%H%M') | int %}
+        {% set current_timestamp = as_timestamp(now) | int %}
 
         {## Get whether date range toggle is enabled as well as start and end date (integer yyyymmdd) ##}
         {## Determine whether current date is within date range using integer (yyyymmdd) comparison ##}
         {% set is_date_range_enabled = is_state('input_boolean.daterange_LOCKNAME_TEMPLATENUM', 'on') %}
-        {% set start_date = states('input_datetime.start_date_LOCKNAME_TEMPLATENUM').replace('-', '') | int %}
-        {% set end_date = states('input_datetime.end_date_LOCKNAME_TEMPLATENUM').replace('-', '') | int %}
-        {% set is_in_date_range = (current_date >= start_date and current_date <= end_date) %}
-    
+        {% set start_date = state_attr('input_datetime.start_date_LOCKNAME_TEMPLATENUM', 'timestamp') | int %}
+        {% set start_time = (states('input_datetime.start_date_LOCKNAME_TEMPLATENUM')[11:]).replace(':', '') | int %}
+        {% set end_date = state_attr('input_datetime.end_date_LOCKNAME_TEMPLATENUM', 'timestamp') | int %}
+        {% set end_time = (states('input_datetime.end_date_LOCKNAME_TEMPLATENUM')[11:]).replace(':', '') | int %}
+
+        {## If the start and end dates match, including time, then we do a basic check against just today ##}
+        {% if (start_date == end_date) %}
```

Comment:
@raman325 is this the check that you're talking about? I can just rip this first one out and make the elif the first if statement, or better (and IMO actually better) just use the final check clause which means that the date and time are relevant always. I personally like this better as it's more what I'm familiar with when it comes to these sort of lock management things.

That last one does mean that the defaults wouldn't work since they always default to same day and 00:00 which is why I actually have this first check in there. For those times when you just want to give someone access for a given day...

Reply:
Yes, I am suggesting the same as you, that we eliminate all of these conditionals and just use the final clause. For the case where a user wants to enable the slot for a single day, I don't think it's a lot to ask them to put in 23:59:59, especially since you have to explicitly enable the date range check for it to be included

Reply:
If we do this, I think we can also get rid of `current_datestamp`, `start_time`, and `end_time` and make this massive logic block just a teensy bit smaller 🙂